### DISCUSSION TITLE
Add getAST() function

### DISCUSSION THREAD

**Discussion by rsmelo92** [ueberdosis/tiptap#2647] [Feature Requests]
It would be really awesome if Tiptap could give support to **export an Abstract Syntax Tree (AST)** as it would let the devs do whatever they need with the result. I know the JSON already helps with that, but an AST is more generic and has a lot of already implementations on it like unifiedJS does (https://github.com/unifiedjs/unified)

Also its good to note that `NuxtJS` created a repo with a helper that does exaclty that, maybe its an alternative to add it to the core code? I would open a PR with that if that helps

https://github.com/nuxtlabs/tiptap-markdown/tree/main/utils

An instant win with this is that tiptap can still never give support to markdown, but everyone that needs it can use the AST to convert to Markdown whenever they need. Win win

**Comment by philippkuehn** [comment]
There is no standard format for an AST so the JSON already is kind of an AST. If you need this in another format you have to do this by yourself for now.
